modules/private_data_fusion_network/private_data_fusion_network/main.tf

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+provider "google" {
+  version = "~> 3.0"
+}
+
+data "google_client_config" "current" {}
+
+resource "google_project_service" "project" {
+  project = var.project_id
+  service = "datafusion.googleapis.com"
+
+  disable_dependent_services = true
+}
+
+resource "google_compute_network" "data_fusion_vpc" {
```

Comment:
In general from what I have seen/been recommended it is better to take things like networks as an input rather than trying to create it in each module that requires it.

@morgante can you take a look and make a recommendation of this is ok?

Otherwise looks good!

Reply:
Usually yes, though it looks like that's what this module is meant for?

Reply:
Right, I guess what I was getting at is whether we should manage the network in this module or just setup the peering with the tenant project and let the user create the network and then provide an example for end-to-end deployment such as https://github.com/terraform-google-modules/terraform-google-sql-db/blob/master/examples/mysql-private/main.tf#L47

Reply:
I see your point my thought was a module to get a network up and configured from scratch but this will be less usable for many customers who already have a network.

Reply:
Changed to take network self link as input variable